src/main/java/csci310group11/Implementation/CollageGenerator.java

```diff
@@ -1,22 +1,109 @@
 package csci310group11.Implementation;
 
-import javq.io.*;
-import java.awt.image.*;
-import javax.imageio.*;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+
+import javax.imageio.ImageIO;
+
+import javafx.scene.layout.BorderImage;
 
 public class CollageGenerator {
-	
-	//download collage to the server
-	public void downloadCollage(Collage collage) {
-		try {
-			BufferedImage image = collage.collage; //getter?
-			String fileName = collage.topic;
-			fileName+= “_” + collage.id  + “.png”;    //if collage has a unique id?
-			File outputFile = new File(topic);
-			ImageIO.write(image, “png”, outputFile);
+	private ArrayList<String> imageUrls; //change to <BufferedImage> if necessary
+	private ArrayList<BufferedImage> images;
+	private ArrayList<BufferedImage> borderedImages;
+	private BufferedImage collage;
+
+	public CollageGenerator(ArrayList<String> urls) {
+		this.imageUrls = new ArrayList<String>();
+		this.images = new ArrayList<BufferedImage>();
+		this.borderedImages = new ArrayList<BufferedImage>();
+		this.collage = new BufferedImage();
```

Comment:
There is no default constructor for BufferedImage initialization--requires params.

Reply:
Yeah. We'll decide on that at the meeting.